src/WebJobs.Extensions/Extensions/Core/Bindings/ExecutionContextBindingProvider.cs

```diff
@@ -60,17 +71,24 @@ public Task<IValueProvider> BindAsync(object value, ValueBindingContext context)
                     throw new ArgumentNullException("context");
                 }
 
-                return BindInternalAsync(CreateContext(context.FunctionInstanceId));
+                return BindInternalAsync(CreateContext(context));
             }
 
-            private ExecutionContext CreateContext(Guid invocationId)
+            private ExecutionContext CreateContext(ValueBindingContext context)
             {
-                return new ExecutionContext
+                var result = new ExecutionContext
                 {
-                    InvocationId = invocationId,
-                    FunctionName = _parameter.Member.Name,
-                    FunctionDirectory = Environment.CurrentDirectory
+                    InvocationId = context.FunctionInstanceId,
+                    FunctionName = context.FunctionContext.MethodName,
+                    FunctionDirectory = Environment.CurrentDirectory,
```

Comment:
Correct.  
We have 2 scenarios: 
1. SDK-only outside of script. Existing behavior here is Env.CurDir. Not sure this is actually useful at all; but I'm leaving that behavior untouched. 
2. Running inside of script. In that case, app dir will always be set (subsequent PR coming), and that will override env.curdir to provide the values we discussed.